### ISSUE TITLE
OpenHours : You want to be able to shift time setting

### ISSUE THREAD

**Issue by miiitaka** [miiitaka/wp-structuring-markup#88]
LocalBusinessでシフト制に対応できるように検討する。
https://wordpress.org/support/topic/office-timings

**Comment by defunctl** [comment]
I'm not sure this will be possible using https://schema.org/OpeningHoursSpecification

I think right now the only way to do it is to switch to using https://schema.org/openingHours

Example:

``` javascript
<script type="application/ld+json">
{
    "@context": "http://schema.org",
    "@type": "LocalBusiness",
    "name": "Sample Name",
    "url": "https://www.sampleurl.com",
    "telephone": "+1-234-567-8900",
    "address": {
        "@type": "PostalAddress",
        "streetAddress": "Sample Address",
        "addressLocality": "Sample City",
        "addressRegion": "Sample Region",
        "postalCode": "Sample Postal",
        "addressCountry": "Sample Country Code"
    },
    "openingHours": [
        "Monday-Friday 08:00-12:00",
        "Monday-Friday 13:00-17:00"
    ]
}
</script>
```

If you agree, I could try to implement it.

**Comment by miiitaka** [comment]
What do you think how to implement input screen?

**Comment by defunctl** [comment]
I was thinking of adding some jQuery to do a UI like this:

![localbusiness-sample](https://cloud.githubusercontent.com/assets/1066195/12411433/77a31114-be33-11e5-8f56-eda6068db377.jpg)

**Comment by miiitaka** [comment]
Well, It is a great idea.
And I am trying to implement, do you give me a pull request?

I'm sorry, English usage is I apologize if I wrong.

**Comment by defunctl** [comment]
If you are doing it already, I can do something else. Or I can just do the interface. 

Let me know

**Comment by miiitaka** [comment]
Yes, I have not done anything yet implemented.

**Comment by defunctl** [comment]
I will try it and give you a pull request. Which version will this be in? 2.4.0?

**Comment by miiitaka** [comment]
Yes, let's release the feature in version 2.4.0!
I have set a milestone in this issue at GitHub.

**Comment by defunctl** [comment]
Assign to me

**Comment by miiitaka** [comment]
I had let me assign you.
Thank you.

**Comment by defunctl** [comment]
Thank you. I will try to get it done as soon as possible!

**Comment by defunctl** [comment]
Almost done this, hopefully finish it over the weekend.

**Comment by defunctl** [comment]
I've implemented an initial version of this https://github.com/miiitaka/wp-structuring-markup/commit/4c49c8d0cba63c5232344e74da688241046fe3bb

I had to change the way the structure was stored in the wp_structuring_markup > local_business table, so we need to clear that on update and tell the user to enter their data again.

**Comment by miiitaka** [comment]
Thank you. Reviewed to have them.

**Comment by defunctl** [comment]
@miiitaka make sure you clear the loca_business column. Can you write something so the plugin does this when it updates, please?

**Comment by miiitaka** [comment]
@defunctl Thank you very much. I appreciate it very much.
To complete the review, it has released a version 2.4.0. Without having to re-update the "LocalBusiness" adds the program so that may be.

**Comment by defunctl** [comment]
Great work on the conversion!
